src/tokens/tokens.service.ts

```diff
@@ -173,16 +162,20 @@ export class TokensService {
     return this.contractAddress;
   }
 
-  private async getStream(ctx: Context) {
-    const stream = this.stream;
+  private async getStream(ctx: Context, namespace: string) {
+    let stream = this.streamCache.get(namespace);
     if (stream !== undefined) {
       return stream;
     }
-    await this.migrationCheck(ctx); // note: may update this.stream
-    const name = this.stream?.name ?? packStreamName(this.topic, this.contractAddress);
-    this.logger.log('Using event stream with name ' + name);
-    this.stream = await this.eventstream.createOrUpdateStream(ctx, name, name);
-    return this.stream;
+    await this.migrationCheck(ctx);
```

Comment:
As in ERC20, I think this should be removed.

Reply:
Actually... looks like you have already repurposed it in this case to do the purge instead of just the check. That's probably fine.